src/debugAdapter/goDebug.ts

```diff
@@ -854,6 +854,21 @@ class GoDebugSession extends LoggingDebugSession {
 			if (goroot && index > 0) {
 				return path.join(goroot, pathToConvert.substr(index));
 			}
+
+			const indexGoModCache = pathToConvert.indexOf(
+				`${this.remotePathSeparator}pkg${this.remotePathSeparator}mod${this.remotePathSeparator}`
+			);
+			const gopath = process.env['GOPATH'];
+
+			if (gopath && indexGoModCache > 0) {
+				return path.join(
+					gopath,
```

Comment:
`gopath` can potentially have multiple paths with the path separator specific to the OS platform. In that case, we can't use it in `path.join` directly this way.

Reply:
Does /pkg/mod apply to the first path in the gopath? Is there a rule around this?
Also, we need to ensure the right path separators are being used like we did in #3108

cc @quoctruong

Reply:
> Does /pkg/mod apply to the first path in the gopath? Is there a rule around this?

Yes. I'm not sure where this is specified, but you can see `goimports` relying on this fact [here](https://github.com/golang/tools/blob/0d839f3cf2ed7d63fd98875ea3589e5bf4b64de7/internal/imports/mod.go#L85).

Reply:
Thanks @stamblerre 

@fnmunhoz, @quoctruong I have pushed a commit to use the first go path, can either of you test this one more time?

Reply:
@ramya-rao-a sure I'll try to do it today and let you know. Thanks.

Reply:
Hey @ramya-rao-a I just double checked the extension after the change you made and it works fine for me, I was able to debug all files, including libraries inside go mod cache.

Thanks!